lib/agent0/agent0/utilities/docker.py

```diff
@@ -0,0 +1,60 @@
+"""Utilities associated to manipulating Docker."""
+import subprocess
+import logging
+import os
+import time
+from pathlib import Path
+
+
+def check_docker(restart: bool = False) -> None:
+    """Check whether docker is running to your liking.
+
+    Parameters
+    ----------
+    restart: bool
+        Restart docker even if it is running.
+    """
+    home_infra = Path(os.path.expanduser("~")) / "code" / "infra"
+    if os.path.exists(home_infra):
+        infra_folder = home_infra
+    else:
+        infra_folder = Path("/code/infra")
+    dockerps = _get_docker_ps_and_log()
```

Comment:
The first thing this should do is check if the caller has docker installed.  something like:
```python
from docker.errors import DockerException

  try:
      client = docker.from_env()
  # Skip this test if docker isn't installed
  except DockerException as exc:
```

Reply:
I also realize you could do this whole thing with the docker module, probably, instead of system calls... I can add a warning to check that docker's installed

Reply:
actually remembered my setup requires another workaround to make the docker module work https://github.com/delvtech/elf-simulations/commit/e3feaf19af566ba5500361111a41e9c4f65f0224

Reply:
added generic check for docker installation including that workaround in ea27a0edcbb15a0c5d5f6b03abc8d562274985bc